tests/src/Traits/OgTrait.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Drupal\joinup\Traits;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\Og;
+use Drupal\og\OgMembershipInterface;
+
+/**
+ * Contains helper methods regarding the organic groups.
+ *
+ * @package src\Traits
+ */
+trait OgTrait {
+
+  /**
+   * User with a given og role is already logged in.
+   *
+   * If the user has more than the required roles, he might have permissions
+   * from the rest of the roles that will lead the test to a false positive.
+   * For this reason, we request check for the specific roles passed.
+   *
+   * @param array $roles
+   *   An array of roles to check.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   A single role, or multiple comma-separated roles in a single string.
+   *
+   * @return bool
+   *   Returns TRUE if the current logged in user has this role (or roles).
+   */
+  protected function loggedInWithOgRole($roles, EntityInterface $entity) {
```

Comment:
I don't see the point of this method. A user doesn't "have an OG role" outside of the context of a particular group. It all depends on the group that is being tested. If the user has this role in group A, but not in group B then this will still return `TRUE` if the test is working in context of group B. This is a timebomb that will result in false positives in the future. This needs at minimum the group passed in.